utils/ProgressBar.go

```diff
@@ -174,11 +174,6 @@ func (pb *ProgressBar) String(termWidth int) string {
 	completed := pb.completedItems // Because this is atomically updated, grab a local reference
 	percentage := float64(completed) / float64(pb.numberItems)
 
-	if percentage != percentage {
```

Comment:
This is still needed as it makes a divide by zero percentage become 100%

Reply:
`percentage != percentage` how does a variable not equal itself?

Reply:
In floating point maths, when you divide something by zero you get `NaN` (Not a Number). `NaN` is never equal to anything, including itself. (https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN)

Thus if you want to check if somebody is NaN, checking if the variable is different to itself will tell you if it is `NaN`. Some languages have a `isNaN` function, so I guess we could swap this out for;

```go
if math.IsNaN(percentage) {
```

Reply:
ah thanks for the explaination :)
Makes sense and I think your suggestion is more readable.